index.js

```diff
@@ -2,45 +2,66 @@ const Entity = require('siren-parser');
 const Action = require('siren-parser/dist/Action');
 const Link = require('siren-parser/dist/Link');
 
+/**
+ * siren-parser wrapper that supports following links and performing actions
+ */
 class Client {
   /**
    * performs an async HTTP request
    * returns object { body, contentType }
    *  body: response body
    *  contentType: Content-Type response header
    *
-   * @callback request
-   * @param {string} uri (required) uri of the resource
+   * @typedef {Object} Response
+   * @property body response body
+   * @property contentType Content-Type response header
+   *
+   * @callback requestFn
+   * @param {string} href (required) href of the resource
    * @param {string} method (required) HTTP request verb ('GET', 'POST', 'PUT', 'PATCH', 'DELETE')
-   * @param {object} fieldValues (optional) field names and values for the request
-   * @return {object} body: response body, contentType: Content-Type response header
+   * @param {Object} [fieldValues] (optional) field names and values for the request
+   * @return {Response} body: response body, contentType: Content-Type response header
```

Comment:
you changed a bunch of the documentation here. Can you explain the changes?
The format of this documentation is defined in jsdocs.
I am not an expert on jsdocs but I thought the thing in the curly braces had to be a type.

Reply:
I was trying to find what are predefined types in JSDoc, and I found this medium post that mentions that suggests to use capitalized names for objects, and only use lower case for boolean, string, and number. I guess also for functions and callbacks. https://medium.com/@trukrs/type-safe-javascript-with-jsdoc-7a2a63209b76

So, in this case type of field values is an object, and type of body is Response, which is defined above. 

Square brackets are to mark values as optional.